commit f5ad52fa9446737902c52b541225a756e1014f72 (refs/changes/22/3022/2)
Author: Alex Wilson <alex.wilson@joyent.com>
Date:   2017-12-01T23:03:30+00:00 (1 year, 10 months ago)
    
    MANTA-3508 invalid user added to "operators" group, mahi sadness ensues
    Reviewed by: Cody Peter Mello <cody.mello@joyent.com>
    Approved by: Cody Peter Mello <cody.mello@joyent.com>

diff --git a/v1/lib/auth_cache.js b/v1/lib/auth_cache.js
index 0db302a..7231340 100644
--- a/v1/lib/auth_cache.js
+++ b/v1/lib/auth_cache.js
@@ -650,6 +650,14 @@ function addGroupMember(self, userdn, groupdn, cb) {
                                   groupdn.indexOf(','));
     var userUuid = userdn.substring(userdn.indexOf('=') + 1,
                                     userdn.indexOf(','));
+
+    if (userUuid === '') {
+        log.warn('ignoring attempt to add invalid user DN "%s" to group %s',
+            userdn, group);
+        cb();
+        return;
+    }
+
     log.info('adding user %s to group %s', userUuid, group);
     // get the login from uuid
     var loginKey = sprintf('/uuid/%s', userUuid);
@@ -670,7 +678,7 @@ function addGroupMember(self, userdn, groupdn, cb) {
                 return;
             }
             log.info({entry: payload}, 'got user entry from redis');
-            payload = JSON.parse(payload);
+            payload = JSON.parse(payload) || {};
             if (!payload.groups) {
                 payload.groups = {};
             }
